### PR TITLE
TDI-37497: tSalesforceOutput logging file content incorrect

### PR DESCRIPTION
**Pull request by jzhao-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
1. Some kind of log generate error
2. When input data with "Id" the insert/upsert would be fails which can work in old framework

**What is the new behavior?**

Fix issue above.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/37/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/37/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/39/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/39/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/42/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/42/)

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 1]
```diff
@@ -221,14 +226,17 @@ private SObject createSObjectForUpsert(IndexedRecord input) {
...
                 } else {
-                    Schema.Field fieldInModule = moduleSchema.getField(se.name());
-                    if (fieldInModule != null) {
-                        // The real type is need in addSObjectField()
-                        addSObjectField(so, fieldInModule.schema().getType(), se.name(), value);
-                    } else {
-                        // This is keep old behavior, when set a field which is not exist.
-                        // It would throw a exception for this.
-                        addSObjectField(so, se.schema().getType(), se.name(), value);
+                    // Make sure to filter the schema "Id", when updating (but not when upserting)
```
Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: misunderstand the comment

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/47/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/47/)

**Pull request merged**